### PR TITLE
{data}[gfbf/2023a] DeltaLake v0.15.1 and fix Arrow v14.0.1 to install pyarrow as an extension

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)

New easyconfig for DeltaLake + update Arrow to properly install `pyarrow` as a `PythonPackage` extension.

Depends on:
* [x] https://github.com/easybuilders/easybuild-easyblocks/pull/3118
* [x] https://github.com/easybuilders/easybuild-easyconfigs/pull/19777

### CODE REVIEW

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot: please test @ generoso
EB_ARGS=" --include-easyblocks-from-pr=3118"

**Comment by lexming** [comment]
@boegelbot: please test @ generoso
EB_ARGS=" --include-easyblocks-from-pr=3118"

**Comment by boegel** [comment]
@boegelbot: please test @ jsc-zen3
EB_ARGS=" --include-easyblocks-from-pr=3118"

**Comment by lexming** [comment]
@boegelbot: please test @ jsc-zen3
EB_ARGS=" --include-easyblocks-from-pr=3118"

File: easybuild/easyconfigs/d/DeltaLake/DeltaLake-0.15.1-gfbf-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,873 @@
...
+    ('zerocopy-derive', '0.7.32'),
+    ('zeroize', '1.7.0'),
+    ('zstd', '0.13.0'),
+    ('zstd-safe', '7.0.0'),
+    ('zstd-sys', '2.0.9+zstd.1.5.5'),
+]
+
+builddependencies = [
+    ('maturin', '1.4.0', '-Rust-1.75.0'),
+    ('Perl', '5.36.1'),  # needed by crate openssl-sys
```
Comment (boegel) [inline easybuild/easyconfigs/d/DeltaLake/DeltaLake-0.15.1-gfbf-2023a.eb:859]: remove trailing whitespace that is tripping up code style check:
```suggestion
    ('Perl', '5.36.1'),  # needed by crate openssl-sys
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@lexming The changes you made to `Arrow` are not working for me (see upcoming test report), why are those required?

It would be better to open a dedicated PR for this if there's bug fix needed there...

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3118
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
node3161.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/88003056c18e08d0a8cb86e71d6591fd for a full test report.

**Comment by branfosj** [comment]
> @lexming The changes you made to `Arrow` are not working for me (see upcoming test report), why are those required?

If I've followed this correctly:
* `DeltaLake` requires `pyarrow>=8`
* The changes to `Arrow-14.0.1-gfbf-2023a.eb` mean that it correctly installs as `14.0.1` instead of `0.0.0`.
* The changes to the Arrow easyconfig require #19651 to make `setuptools_scm` available.
* 19651 targeted `5.0.x` and this PR targets `develop`

This means that merging the changes to `Arrow` will cause it to be broken in `develop` until we collapse in the `5.0.x` branch.

**Comment by lexming** [comment]
@branfosj that's correct, changes to `Arrow` in this PR make it to properly install as a `PythonPackage` extension which triggers extra checks such as `pip check`. This will capture the wrong `0.0.0` version unless a Python with `setuptools_scm` is used.
I forgot that we put the changes on `setuptools_scm` into `5.0` (I have it deployed in my dev system already :yum: ). Maybe we should move it to `develop` to avoid perpetuating such issues for too long.

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot: please test @ generoso
EB_ARGS=" --include-easyblocks-from-pr=3118"

**Comment by lexming** [comment]
@boegel Python fixed in `develop`, tests from the bots are fine. Please try again.

**Comment by Micket** [comment]
Test report by @Micket
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3118
**SUCCESS**
Build succeeded for 3 out of 3 (2 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/Micket/b22b34c9af98456b35166f15d02361d0 for a full test report.

**Review by Micket — APPROVED** [review]
lgtm

**Comment by Micket** [comment]
Going in, thanks @lexming!

**Pull request merged**